plugins/plugin-chart-echarts/src/Timeseries/buildQuery.ts

```diff
@@ -32,18 +38,23 @@ export default function buildQuery(formData: QueryFormData) {
           : timeseries_limit_metric
           ? [[timeseries_limit_metric, !order_desc]]
           : [],
+        time_offsets: isValidTimeCompare(formData, baseQueryObject) ? formData.time_compare : [],
         post_processing: [
-          {
-            operation: 'pivot',
-            options: {
-              index: ['__timestamp'],
-              columns: formData.groupby || [],
-              // Create 'dummy' sum aggregates to assign cell values in pivot table
-              aggregates: Object.fromEntries(
-                metricLabels.map(metric => [metric, { operator: 'sum' }]),
-              ),
-            },
-          },
+          timeCompareOperator(formData, baseQueryObject),
+          rollingWindowOperator(formData, baseQueryObject),
+          isValidTimeCompare(formData, baseQueryObject)
```

Comment:
Instead of extracting one operator at a time, I'm wondering whether it makes more sense to create a monolith `extractPostProcessing` function as these operator factories seem to have a lot of duplicate code?

It can be part of the standard `buildQueryObject` process, similar to `extractQueryFields`, so it can be used by other charts as well.

In general, we should avoid special chart-specific `buildQuery` logics as much as possible.

Reply:
I should be doing some iterations in the future. Make these post processing more generic.